File: backend/models/Message.js
const pool = require('../config/database');

class Message {
  static async create(conversationId, senderId, message) {
    const [result] = await pool.execute(
      'INSERT INTO messages (conversation_id, sender_id, message, sent_at) VALUES (?, ?, ?, NOW())',
      [conversationId, senderId, message]
    );
    return result.insertId;
  }

  static async findByConversationId(conversationId) {
    const [rows] = await pool.execute(
      `SELECT m.*, u.email as sender_email, u.role as sender_role,
       COALESCE(sp.full_name, tpv.full_name, u.email) as sender_name
       FROM messages m
       JOIN users u ON m.sender_id = u.id
       LEFT JOIN student_profiles sp ON sp.user_id = u.id
       LEFT JOIN teacher_profiles tpv ON tpv.user_id = u.id
       WHERE m.conversation_id = ?
       ORDER BY m.sent_at ASC`,
      [conversationId]
    );
    return rows;
  }

  static async markAsRead(conversationId, userId) {
    await pool.execute(
      'UPDATE messages SET is_read = TRUE WHERE conversation_id = ? AND sender_id != ? AND is_read = FALSE',
      [conversationId, userId]
    );
  }
}

module.exports = Message;

File: tutionmediaapp/lib/models/message.dart
class Message {
  final int id;
  final int conversationId;
  final int senderId;
  final String message;
  final String sentAt;
  final String? senderEmail;
  final String? senderName;
  final String? senderRole;

  Message({
    required this.id,
    required this.conversationId,
    required this.senderId,
    required this.message,
    required this.sentAt,
    this.senderEmail,
    this.senderName,
    this.senderRole,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['id'],
      conversationId: json['conversation_id'],
      senderId: json['sender_id'],
      message: json['message'],
      sentAt: json['sent_at'],
      senderEmail: json['sender_email'],
      senderName: json['sender_name'],
      senderRole: json['sender_role'],
    );
  }

  get content => null;

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'conversation_id': conversationId,
      'sender_id': senderId,
      'message': message,
      'sent_at': sentAt,
    };
  }
}